src/Functions/IFunction.cpp

```diff
@@ -310,7 +311,10 @@ ColumnPtr IExecutableFunction::executeWithoutSparseColumns(const ColumnsWithType
 
 ColumnPtr IExecutableFunction::execute(const ColumnsWithTypeAndName & arguments, const DataTypePtr & result_type, size_t input_rows_count, bool dry_run) const
 {
-    if (useDefaultImplementationForSparseColumns())
+    bool use_default_implementation_for_sparse_columns = useDefaultImplementationForSparseColumns();
+    /// DataTypeFunction does not supports obtaining default (isDefaultAt())
+    /// ColumnFunction does not uspport getting specific values
```

Comment:
A typo: `uspport`

Why don't return a tuple of captured arguments as a value?

Reply:
I don't see how it is interchangeable - https://github.com/ClickHouse/ClickHouse/blob/b38d4b5b0fb6956242e9a870ea5d6275210f3a22/src/Functions/IFunction.cpp#L361-L368

`canBeInsideSparseColumns` can be used here, but there is special handling of const result, so I decided not to change this.